File: Codeforces/round714/c.py
# solve
def solve():
    import sys
    input = sys.stdin.readline
    mod = 1000000007
    t = int(input())
    dp = [None] * (2 * 10 ** 5 + 10)
    dp[0] = [0] * 10
    dp[0][0] = 1
    dpi = dp[0]
    dpc = [0] * (2 * 10 ** 5 + 10)
    dpc[0] = 1
    for i in range(2 * 10 ** 5 + 9):
        dpi1 = [dpi[9]] + dpi[:-1]
        dpi1[1] = (dpi[0] + dpi[9]) % mod
        dpc[i + 1] = (dpc[i] + dpi[9]) % mod
        dpi = dpi1
        dp[i + 1] = dpi1
    alis = [None] * t
    for i in range(t):
        n, m = input().split()
        m = int(m)
        ans = 0
        for ni in n:
            ans += dpc[m + int(ni)]
            if ans >= mod:
                ans %= mod
        alis[i] = str(ans)
    print("\n".join(alis))
    return
# main
if __name__ == '__main__':
    solve()

File: Codeforces/round714/c.cpp
#include <iostream>

using namespace std;
# Define N 200000
int main(int argc, char const *argv[])
{
    long int mod = 1000000007;
    int t, n, m;
    cin >> t;
    long int dp[N + 11][10] = {};
    long int dpc[N + 11] = {};
    dp[0][0] = 1;
    for (int i = 0; i < N + 10; i++)
    {
        dp[i + 1][0] = dp[i][9];
        dp[i + 1][1] = dp[i][9];
        for (int j = 0; j < 9; j++)
        {
            dp[i + 1][j + 1] += dp[i][j];
            dp[i + 1][j + 1] %= mod;
        }
        for (int j = 0; j < 10; j++)
        {
            dpc[i] += dp[i][j];
            dpc[i] %= mod;
        }
    }
    for (int i = 0; i < t; i++)
    {
        scanf("%d%d", &n, &m);
        long int x;
        long int ans = 0;
        while (n != 0)
        {
            x = n % 10;
            ans += dpc[m + x];
            ans %= mod;
            n /= 10;
        }
        printf("%ld\n", ans);
    }

    return 0;
}